src/components/button/button.scss

```diff
@@ -655,15 +656,26 @@
   --calcite-button-padding-x: 15px;
   @apply text-0h;
 }
-:host([scale="l"]:not([appearance="transparent"])) button.content--slotted,
-:host([scale="l"]:not([appearance="transparent"])) a.content--slotted {
-  --calcite-button-padding-y: 11px;
+
+:host([scale="l"]:not([appearance="transparent"])) {
+  .button-padding {
+    --calcite-button-padding-y: 11px;
+  }
+  //shrink the padding if icon is present to preserve the height
+  .button-padding-shrunk {
```

Comment:
nitpick: I think the selector for this ideally would be `button-padding--shrunk`. using BEM naming methodology.  https://getbem.com/